tfutils/base.py

```diff
@@ -97,83 +98,131 @@ def __init__(self,
-        self.database = self.conn[dbname]
-        self.collfs = gridfs.GridFS(self.database, collname)
-        self.coll = self.collfs._GridFS__files
-        recent_name = '_'.join([dbname, collname, exp_id, '__RECENT'])
+        self.tfsaver_args = tfsaver_args
+        self.tfsaver_kwargs = tfsaver_kwargs
+        
+        if save_params is None:
+            save_params = {}
+        if load_params is None:
+            load_params = {}
+        location_variables = ['host', 'port', 'dbname', 'collname', 'exp_id']
+        for _k in location_variables:
+            if _k in save_params:
+                sv = save_params[_k]
+            else:
+                sv = load_params[_k]
+            if _k in load_params:
+                lv = load_params[_k]
+            else:
+                lv = save_params[_k]
+            setattr(self, _k, sv)
+            setattr(self, 'load_' + _k, lv)
+        self.sameloc = all([getattr(self, _k) == getattr(self, 'load_' + _k) for _k in location_variables] )
+
+        for _k in ['do_save', 'save_metrics_freq', 'save_valid_freq', 'cache_filters_freq',
+                   'save_filters_freq', 'save_initial', 'save_to_gfs']:
+            setattr(self, _k, save_params.get(_k, DEFAULT_SAVE_PARAMS[_k]))
+
+        for _k in ['do_restore']:
+            setattr(self, _k, load_params.get(_k, DEFAULT_LOAD_PARAMS[_k]))
+
+        self.conn = pymongo.MongoClient(host=self.host, port=self.port)
+        self.collfs = gridfs.GridFS(self.conn[self.dbname], self.collname)
+        recent_name = '_'.join([self.dbname, self.collname, self.exp_id, '__RECENT'])
         self.collfs_recent = gridfs.GridFS(self.conn[recent_name])
-        self.coll_recent = self.collfs_recent._GridFS__files
 
-        if tensorboard_dir is not None:  # save graph to tensorboard
-            tf.train.SummaryWriter(tensorboard_dir, tf.get_default_graph())
+        self.load_data = None     
+        load_query = load_params.get('query')
+        if load_query is None:
+            load_query = {}
+        else:
+            if self.sameloc:
+                raise Exception('Loading pointlessly')
+        load_query.update({'exp_id': self.load_exp_id})
+        self.load_query = load_query       
+        if self.load_host != self.host or self.port != self.load_port:
+            self.load_conn = pymongo.MongoClient(host=self.load_host,
+                                                     port=self.load_port)
+        else:
+            self.load_conn = self.conn
+        self.load_collfs = gridfs.GridFS(self.load_conn[self.load_dbname], 
+                                             self.load_collname)
+        load_recent_name = '_'.join([self.load_dbname,
+                                     self.load_collname,
+                                     self.load_exp_id,
+                                     '__RECENT'])
+        self.load_collfs_recent = gridfs.GridFS(self.load_conn[load_recent_name])
```

Comment:
Sorry, I meant, "do the same."    If we're going to construct self.collfs and self.collfs_recent in the initializer, let's do the same for the load_ stuff, since they're of the same "importance" and serve a similar function.  We shouldn't split up the initialization of the two sets of variables.